examples/Substrait.ipynb

```diff
@@ -0,0 +1,254 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import duckdb\n",
+    "import os\n",
+    "from urllib.request import urlretrieve\n",
+    "\n",
+    "\n",
+    "class DatabaseServer:\n",
+    "    DB_NAME = \"palmer_penguins.ddb\"\n",
+    "    DB_URL = \"https://storage.googleapis.com/ibis-tutorial-data/palmer_penguins.ddb\"\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        if not os.path.exists(self.DB_NAME):\n",
+    "            urlretrieve(self.DB_URL, self.DB_NAME)\n",
+    "        self.db = duckdb.connect(self.DB_NAME)\n",
+    "        self.db.install_extension(\"substrait\")\n",
+    "        self.db.load_extension(\"substrait\")\n",
+    "    \n",
+    "    def execute(self, substrait):\n",
+    "        result = self.db.from_substrait(substrait)\n",
+    "        return result.fetchall()\n",
+    "    \n",
+    "    @property\n",
+    "    def connection_string(self):\n",
+    "        \"\"\"Helper for our toy example\"\"\"\n",
+    "        return self.DB_NAME\n",
+    "        \n",
+    "\n",
+    "db_server = DatabaseServer()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Ibis Table\n",
+    "\n",
+    "We need an Ibis Table to query against. We'll take a shortcut in our example by querying our local DB directly. \n",
+    "\n",
+    "Since we only need the table name and schema, we acquire the table through our backend connection and then call [unbind()](https://ibis-project.org/api/expressions/?h=unbind#ibis.expr.types.core.Expr.unbind). This will remove any backend-specific objects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ibis\n",
+    "\n",
+    "con = ibis.duckdb.connect(db_server.connection_string)\n",
+    "table = con.table(\"penguins\").unbind()\n",
+    "print(table)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Substrait Compiler\n",
+    "\n",
+    "The `ibis-substrait` package provides a `SubstraitCompiler` that can both compile and decompile Substrait Plans.\n",
+    "\n",
+    "Let's see it in action:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ibis import _\n",
+    "from ibis_substrait.compiler.core import SubstraitCompiler\n",
+    "\n",
+    "compiler = SubstraitCompiler()\n",
+    "\n",
+    "query = (\n",
+    "    table\n",
+    "    .select(_.species)\n",
+    "    .group_by(_.species)\n",
+    "    .agg(count=_.species.count())\n",
+    ")\n",
+    "\n",
+    "substrait_plan = compiler.compile(query)\n",
+    "\n",
+    "print(substrait_plan)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The object created is an instantiation of a Python class generated from the Substrait protocol buffers defined by the Substrait specification."
```

Comment:
I'm not sure we need this level of explanation?

Reply:
Agreed! Will remove